IdentityCore/src/workplacejoin/ios/MSIDWorkPlaceJoinUtil.m

```diff
@@ -29,11 +29,68 @@
 #import "MSIDWorkplaceJoinChallenge.h"
 #import "MSIDWorkPlaceJoinUtilBase+Internal.h"
 
+static NSString *kWPJPrivateKeyIdentifier = @"com.microsoft.workplacejoin.privatekey\0";
+
 @implementation MSIDWorkPlaceJoinUtil
 
-+ (MSIDAssymetricKeyPairWithCert *)getWPJKeysWithContext:(id<MSIDRequestContext>)context
++ (MSIDWPJKeyPairWithCert *)getWPJKeysWithTenantId:(NSString *)tenantId context:(id<MSIDRequestContext>)context
 {
-    return [self getRegistrationInformation:context workplacejoinChallenge:nil];
+    NSString *teamId = [[MSIDKeychainUtil sharedInstance] teamId];
+    
+    if (!teamId)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, context, @"Encountered an error when reading teamID from keychain.");
+        return nil;
+    }
+    
+    NSString *legacySharedAccessGroup = [NSString stringWithFormat:@"%@.com.microsoft.workplacejoin", teamId];
+    NSData *tagData = [kMSIDPrivateKeyIdentifier dataUsingEncoding:NSUTF8StringEncoding];
+    
+    NSDictionary *extraPrivateKeyAttributes = @{ (__bridge id)kSecAttrApplicationTag: tagData,
+                                                 (__bridge id)kSecAttrAccessGroup : legacySharedAccessGroup };
+    NSDictionary *extraCertAttributes = @{ (__bridge id)kSecAttrAccessGroup : legacySharedAccessGroup };
+    
+    MSIDWPJKeyPairWithCert *legacyKeys = [self findWPJRegistrationInfoWithAdditionalPrivateKeyAttributes:extraPrivateKeyAttributes certAttributes:extraCertAttributes context:context];
+        
+    if (legacyKeys)
+    {
+        if ([NSString msidIsStringNilOrBlank:tenantId])
+        {
+            // ESTS didn't request a specific tenant, just return default one
+            return legacyKeys;
+        }
+        
+        // Read tenantId for legacy identity
+        NSString *registrationTenantId = [MSIDWorkPlaceJoinUtil getWPJStringDataForIdentifier:kMSIDTenantKeyIdentifier context:context error:nil];
```

Comment:
nit: This also populates error if registrationTenantId is blank. Maybe can check for error to avoid checking if registrationTenantId is nilOrBlank again

Reply:
Addressed